### DISCUSSION TITLE
Release schedule for jOOQ?

### DISCUSSION THREAD

**Discussion by atollk** [jOOQ/jOOQ#14518] [Q&A] [answered]
Is there an official release schedule for new minor versions of jOOQ? Specifically in my case, I am waiting for new features contained in 3.18, but a general answer, if possible, would also be appreciated.

**Comment by lukaseder** [comment, marked as answer]
There is none. The tradeoff of equally bad approaches isn't clearly in favour of one or the other:

- Time based schedules suffer from the problem that stuff that isn't finished will be shipped nonetheless
- Feature based schedules suffer from the problem that they keep being delayed

In short, both approaches just produce unkept promises and unhappy users, *or* stressed out maintainers. For no good and obvious reason. Hence, jOOQ simply doesn't have schedules.

Note you can always access the snapshots from here: https://www.jooq.org/download/versions

In the short term, 3.18 is "under way." Most major features have been implemented, hoping to release in Q1 2023 (but of course: no promises). The major features being:

- Much more QOM implementation
- Tons of DiagnosticsListener improvements and new SQL transformations
- More SQL/JSON support
- Oracle associative array support
- PostgreSQL Multi dimensional array types
- R2DBC 1.0 support
